core/thread/inc/ROOT/TThreadedObject.hxx

```diff
@@ -175,6 +205,9 @@ namespace ROOT {
       {
          if ( i >= fObjPointers.size()) {
             Warning("TThreadedObject::GetAtSlot", "Maximum number of slots reached.");
+            if (fHadIMTDisabled && ROOT::GetImplicitMTPoolSize() > fObjPointers.size())
+               Warning("TThreadedObject::GetAtSlot",
+                  "To use TThreadedObject with IMT, turn IMT on before constructing a TThreadedObject.");
```

Comment:
due to TBB scheduling, it's almost certain that the program will crash or otherwise have unexpected behavior if `ROOT::GetImplicitMTPoolSize() > fObjPointers.size()`. I would suggest something harsher than a warning here, it can't hurt.

Reply:
Unless the `TThreadedObject` isn't used with IMT! Which is why I want to keep it a warning. Think
```
TThreadedObject obj;
// do something with std::threads
...
ROOT::EnableImplicitMT();
...
auto ptr = obj.Merge();
if (*ptr != obj.GetAtSlot(64))...
```

Reply:
_fiiiine_ :P